com.unity.render-pipelines.universal/Editor/ShaderPreprocessor.cs

```diff
@@ -41,20 +41,33 @@ internal class ShaderPreprocessor : IPreprocessShaders
         ShaderKeyword m_CascadeShadows = new ShaderKeyword(ShaderKeywordStrings.MainLightShadowCascades);
         ShaderKeyword m_SoftShadows = new ShaderKeyword(ShaderKeywordStrings.SoftShadows);
         ShaderKeyword m_MixedLightingSubtractive = new ShaderKeyword(ShaderKeywordStrings.MixedLightingSubtractive);
-        ShaderKeyword m_Lightmap = new ShaderKeyword("LIGHTMAP_ON");
-        ShaderKeyword m_DirectionalLightmap = new ShaderKeyword("DIRLIGHTMAP_COMBINED");
-        ShaderKeyword m_AlphaTestOn = new ShaderKeyword("_ALPHATEST_ON");
-        ShaderKeyword m_GbufferNormalsOct = new ShaderKeyword("_GBUFFER_NORMALS_OCT");
+        ShaderKeyword m_Lightmap = new ShaderKeyword(ShaderKeywordStrings.LIGHTMAP_ON);
+        ShaderKeyword m_DirectionalLightmap = new ShaderKeyword(ShaderKeywordStrings.DIRLIGHTMAP_COMBINED);
+        ShaderKeyword m_AlphaTestOn = new ShaderKeyword(ShaderKeywordStrings._ALPHATEST_ON);
+        ShaderKeyword m_GbufferNormalsOct = new ShaderKeyword(ShaderKeywordStrings._GBUFFER_NORMALS_OCT);
         ShaderKeyword m_UseDrawProcedural = new ShaderKeyword(ShaderKeywordStrings.UseDrawProcedural);
         ShaderKeyword m_ScreenSpaceOcclusion = new ShaderKeyword(ShaderKeywordStrings.ScreenSpaceOcclusion);
 
+        ShaderKeyword m_LocalDetailMulx2;
+        ShaderKeyword m_LocalDetailScaled;
+        ShaderKeyword m_LocalClearCoat;
+        ShaderKeyword m_LocalClearCoatMap;
+
         int m_TotalVariantsInputCount;
         int m_TotalVariantsOutputCount;
 
         // Multiple callback may be implemented.
         // The first one executed is the one where callbackOrder is returning the smallest number.
         public int callbackOrder { get { return 0; } }
 
+        void InitializeLocalShaderKeywords(Shader shader)
+        {
+            m_LocalDetailMulx2 = new ShaderKeyword(shader, ShaderKeywordStrings._DETAIL_MULX2);
```

Comment:
Actually, what does MULX2 mean? `*2`? `*x^2`? what is it multiplying?

Reply:
This is naming that comes from @lukaschod Detail implementation. I believe it's the same as Built-In uses. 
Not saying it's very descriptive but this is not in the scope of this PR.

Reply:
It indicates that detail color is multiplied by 2. We treat 0.5 color as white.